src/evaluation/WorkspaceManager.jl

```diff
@@ -224,57 +203,52 @@ end
 "Call `cd(\$path)` inside the workspace. This is done when creating a workspace, and whenever the notebook changes path."
 function cd_workspace(workspace, path::AbstractString)
     eval_in_workspace(workspace, quote
-        cd($(path |> dirname))
+        cd(dirname($path))
     end)
 end
 
 "Create a new empty workspace. Return the `(old, new)` workspace names as a tuple of `Symbol`s."
 function bump_workspace_module(session_notebook::SN)
     workspace = get_workspace(session_notebook)
     old_name = workspace.module_name
-    new_name = workspace.module_name = create_emptyworkspacemodule(workspace.pid)
+    new_name = workspace.module_name = create_emptyworkspacemodule(workspace.worker)
 
     old_name, new_name
 end
 
 function possible_bond_values(session_notebook::SN, n::Symbol; get_length::Bool=false)
     workspace = get_workspace(session_notebook)
-    pid = workspace.pid
 
-    Distributed.remotecall_eval(Main, pid, quote
+    Malt.remote_eval_fetch(workspace.worker, quote
         PlutoRunner.possible_bond_values($(QuoteNode(n)); get_length=$(get_length))
     end)
 end
 
-function create_emptyworkspacemodule(pid::Integer)::Symbol
-    Distributed.remotecall_eval(Main, pid, :(PlutoRunner.increment_current_module()))
+function create_emptyworkspacemodule(worker::Malt.Worker)::Symbol
+    Malt.remote_eval_fetch(worker, quote
+        PlutoRunner.increment_current_module()
+    end)
 end
 
-const Distributed_expr = :(
-    Base.loaded_modules[Base.PkgId(Base.UUID("8ba89e20-285c-5b6f-9357-94700520ee1b"), "Distributed")]
-)
-
 # NOTE: this function only start a worker process using given
 # compiler options, it does not resolve paths for notebooks
 # compiler configurations passed to it should be resolved before this
-function create_workspaceprocess(;compiler_options=CompilerOptions())::Integer
-    # run on proc 1 in case Pluto is being used inside a notebook process
-    # Workaround for "only process 1 can add/remove workers"
-    pid = Distributed.remotecall_eval(Main, 1, quote
-        $(Distributed_expr).addprocs(1; exeflags=$(_convert_to_flags(compiler_options))) |> first
-    end)
+function create_workspaceprocess(;compiler_options=CompilerOptions())::Malt.Worker
+    worker = Malt.Worker(;exeflags=_convert_to_flags(compiler_options)) # FIXME(savq): exeflags
```

Comment:
```suggestion
    worker = Malt.Worker(;exeflags=_convert_to_flags(compiler_options)) # TODO(savq): exeflags
```
Let's stick to TODO? Otherwise, people have to search for `TODO`, `FIXME`, and possibly other comments to find tasks. (Even better would be to track the task in an issue for better visibility.)

Reply:
Sure. I'll make sure to change the other comments.

You can even see that what this comment is referring to (the flags) was already fixed 🙈

Reply:
No problem :+1: Happens to everyone